src/main/java/org/broadinstitute/hellbender/tools/copynumber/CondenseReadCounts.java

```diff
@@ -0,0 +1,203 @@
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.CopyNumberProgramGroup;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.copynumber.formats.collections.SimpleCountCollection;
+import org.broadinstitute.hellbender.tools.copynumber.formats.records.SimpleCount;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import org.broadinstitute.hellbender.utils.tsv.DataLine;
+import org.broadinstitute.hellbender.utils.tsv.TableColumnCollection;
+import org.broadinstitute.hellbender.utils.tsv.TableReader;
+import org.broadinstitute.hellbender.utils.tsv.TableWriter;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@CommandLineProgramProperties(summary = "condenses read-counts in .tsv format into a new file also in .tsv format",
+        oneLineSummary = "condenses read-counts in .tsv format into a new file also in .tsv formas" ,
+        programGroup = CopyNumberProgramGroup.class)
+public class CondenseReadCounts extends CommandLineProgram {
+
+    public static final String FACTOR_LONG_NAME = "factor";
+    public static final String FACTOR_SHORT_NAME = "f";
+    public static final String EXPECTED_BIN_LENGTH_LONG_NAME = "out-bin-length";
+
+    @Argument(
+            doc = "Path to the input counts file",
+            fullName = StandardArgumentDefinitions.INPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.INPUT_SHORT_NAME
+    )
+    private String input;
+
+    @Argument(
+            doc = "Path to the output counts file",
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME
+    )
+    private String output;
+
+    @Argument(
+            doc = "Indicates how many input read count record must be condense into each output record",
+            fullName = FACTOR_LONG_NAME,
+            shortName = FACTOR_SHORT_NAME,
+            minValue = 1
+    )
+    private int factor = 0;
+
+    @Argument(
+            doc = "expected bin-length after condensation, those output bin that do not match this length will be excluded. " +
+                    "The default 0 means that such constrain is not be applied",
+            fullName = EXPECTED_BIN_LENGTH_LONG_NAME,
+            minValue = 0,
+            optional = true
+    )
+    private int expectedBinLength = 0;
+
+    @Override
+    public String doWork() {
+
+        try (final SimpleCountReader reader = new SimpleCountReader(input);
+             final SimpleCountWriter writer = new SimpleCountWriter(output)) {
+            String contig = null;
```

Comment:
This logic needs tests and should be extracted.  A precise description of the method would be nice.

That said, I think some thought might be needed to nail down the method we want, just to make sure filtering/blacklisting doesn't trip us up.

Reply:
Will try to make the code a bit more readable and add documentation... but it is quite simple what it does... it is the more simple implementation of condensing adjacent counts i.e. greedily from up-to-downstream. Basically its documentation would be in the description of the tool.

Reply:
If the Javadoc is sufficiently precise, this is fine.  Or you can have more user friendly and breezy Javadoc and leave the details to comments here or the backend method documentation.